### ISSUE TITLE
OXP Mini 5800U HandyGCCS Button Mapping Not Working 

### ISSUE THREAD

**Issue by Derainian** [holoiso-eol/holoiso#376]
Using a OneXPlayer Mini 5800U. When using the 5.18 kernal boot option some of my button mapping from HandyGCCS does not work. It only seems to work in the 5.13 neptune kernal boot. Reinstalled HandyGCCS and it still will not map them. The following buttons have lost functionality in 5.18:

- Home button (does not work for either quick or long press functions)
- Volume rocker 
- Keyboard button (only allows switching to mouse mode, quick pressing will not bring up keyboard)

I would use the 5.13 neptune version if I could get the wifi card working on it, but after reading all the relevant posts that doesn't seem to be an option for anything older than 5.16. Hoping there's a quick fix for this, 5.18 is usable for the most part without these buttons, but without the home button mapping I can't use the quick access menu to play with performance settings.

**Comment by jxu91** [comment]
Do you know how to fix the black screen problem on OXP Mini 5800U? I have to plug in a monitor whenever I boot into the 5.18 kernel, while the Neptune kernel has the Wifi issue.
Thanks!

**Comment by Derainian** [comment]
Unfortunately I am experiencing the same issue with the black screen on 5.18. I was planning to open a separate thread for it if I couldn't find one specific to the OXP Mini 5800U use case. My best guess as to what is causing it is some sort of initialization issue when trying to access the screen's driver at bootup. It could have something to do with the screen's driver not being accessed properly by the 5.18 kernal on bootup. I think plugging in the external screen could be causing the screen to briefly restart and call its driver successfully the second time. Again though, this is just my best guess, so I could be wrong. It might be worth looking at though. I haven't had the time to go poking into the OS files to compare the information between boot options either, but I will try to look at them when I get the chance.

When investigating the issue further, I saw some people reporting that if they left it sitting long enough on the black screen that it would eventually show up. I have not been able to replicate their findings, so I cannot say that it is a viable workaround like plugging in the external display.

**Comment by alansari** [comment]
https://GitHub.com/ruineka/gamescope-onexplayer
Follow the instructions to fix the black screen and it includes the handygccs.git in the setup. 

No issues running the 5.18 kernel nor with gamescope without a little tweaking.

**Comment by Derainian** [comment]
The problem here is that I used ruineka's fixes to map my buttons in the first place. I figured I could have done something wrong when running it though, so I did a fresh install of holoiso and followed them again to be safe. 

I got the buttons working on 5.18 now. I think the problem before was that I forgot to install the dependencies first. Ruineka's fixes still haven't fixed my black screen booting for 5.18 though. I don't know what else I could have done differently to fix the black screen bootup. I watched ruineka's YouTube videos and they said that the black screen boot seems to happen to some, but not all onexplayer minis. Did you have the blackscreen boot before, but fixed it with ruineka's fixes, or were you lucky enough that yours just always worked?

**Comment by jxu91** [comment]
I also used ruineka's fixes and the button map works, however the black screen issue still exists. The ChimeraOS works fine with ruineka's fixes. Actually I met the black screen issue while installing HoloISO, and only plug in a monitor could make the screen work. The old command line installing and the new beta GUI installing all have the same issue, suggesting the root of issue is neither gamescope, nor desk manager, but the kernel of HoloISO.

**Comment by alansari** [comment]
Edit /usr/bin/gamescope-session make sure the output is set to -O '*',eDP-1

**Comment by Derainian** [comment]
![IMG_20220720_103353215](https://user-images.githubusercontent.com/102991836/180036594-d25dc2ca-0260-430a-85fb-e94868710854.jpg)
Looking through the gamescope-session file. The only line I found with the code piece you gave us is in the picture I attached, but I'm guessing that isn't the line I need to change. In the interest of not royally messing this up, which line number in the code has the output I need to change?

**Comment by alansari** [comment]
Try removing the '*' so it looks like -o eDP-1. If that doesn't fix it I can't remember exactly what I did to resolve the issue... You can also go take the gamescope session from chimeraos GitHub. I recall doing that so I could use environment.d for overrides.

**Comment by Derainian** [comment]
I gave it a shot and all it did was make my external display unusable unless I booted the system with it plugged in. Still got the black screen bootup. I went to revert the change and for some reason it just bricked its boot, so I had to do another fresh holoiso install. I'm still open to ideas here if you remember how you did it. Until then, I'll keep fiddling with some possible fix ideas.

**Comment by jxu91** [comment]
It's true that the output of gamescope-session on chimeraOS is -O /'/*/', eDP-1 ,a little different from the one on HoloISO, but even I cloned the whole gamescope-session file to HoloISO didn't help. Also, after running holoiso-disable-session which disabled the startup running of gamescope-session, the black screen issue still existed. So the screen went wrong before gamescope-session worked.  That's what I've tried before and I have to transmit back to ChimeraOS now. The HoloISO is not usable on the 5800U mini unless this black screen issue is solved.

**Comment by Derainian** [comment]
I've still been fiddling with mine to see what works. I tried writing a login script to force the display to restart, but that didn't seem to do anything. Comically enough though, I found that if you aren't concerned about having a login screen, you can still use all of the buttons while the screen is black. If you do the following, you can force the display to work by navigating to desktop mode:

1. Do a long enough press on the home button. You should hear the slideout tone when you release the button after pressing it for about 2-3 seconds.
2. Press up on the d-pad once and press A
3. Press up on the d-pad twice and press A

After following those steps, you should be forced into desktop mode, which will correct your display. Once you click on the "Return to Gaming Mode" icon it should take you back to game mode without any issues.

Obviously this is a bandaid fix for the issue, but it will at least make the system usable enough on the OXP Mini 5800U for the time being. I think someone far more skilled at Linux than me ought to look into why the 5.18 kernel is behaving the way it is so we can see a true permanent solution in the future. I will still toy around with my system if I think of any other ideas.

**Comment by pastaq** [comment]
Black screen issue is a BIOS issue. We are working with OXP for a fix. For now the workaround of plugging in to an external display is the only solution available. New BIOS are being tested though, so a solution is relatively near. I can't give an ETA though.

If you're still having button issues, please open an issue on the HandyGCCS git so I can solve any issues directly. From the comments it seems like you solved it though.

**Comment by Derainian** [comment]
Indeed. The button issue is no longer a problem save for one odd thing I noticed. I don't know if it is a big enough issue to open a new thread on, but I noticed that if I have my OXP mini set to mouse mode when I boot the image up or return to game mode from desktop, there is a decent chance that the buttons will not respond after turning mouse mode off. It isn't super inconvenient, as the fix is as easy as switching to desktop mode and back again with mouse mode turned off, but I thought I'd mention it since I noticed it.
